eclass/haskell-cabal.eclass

```diff
@@ -528,10 +533,12 @@ cabal_src_compile() {
 				cabal-haddock
 			fi
 		fi
-		# re-run build, to recover stuff that haddock makes disappear for some
-		# reason...thankfully, cabal is smart enough to only re-build the parts
-		# that have been lost, so this isn't too expensive.
-		cabal-build
+		if [[ -n "${CABAL_DOCTEST_FIX}" ]]; then
+			# re-run build, to recover stuff that haddock makes disappear for some
```

Comment:
I also suggest adding `einfo` (or `ewarn`) here that describes enabled workaround.